content/en/blog/_posts/2023-08-04-kubernetes-1.28-blog.md

```diff
@@ -0,0 +1,290 @@
+
+This adds "match conditions" to admission webhooks, as an extension to the existing rules to define the scope of a webhook. A matchCondition is a CEL expression that must evaluate to true for the admission request to be sent to the webhook. If a matchCondition evaluates to false, the webhook is skipped for that request (implicitly allowed).
+    
+ValidatingAdmissionPolicy is an exciting new feature that we hope will greatly reduce the need for admission webhooks, but it is intentionally not attempting to cover every possible use case. This proposal aims to improve the situation for those webhooks that cannot be migrated.
+    
+## Unknown Version Interoperability Proxy
+
+When a cluster has multiple API servers at mixed versions (such as during an upgrade/downgrade or when runtime-config changes and a rollout happens), not every apiserver can serve every resource at every version.
+
+To fix this, the filter is added to the handler chain in the aggregator which proxies clients to an API server that is capable of handling their request.
+    
+When an upgrade or downgrade is performed on a cluster, for some period of time the api servers are at differing versions and are able to serve different sets of built-in resources (different groups, versions, and resources are all possible).
+
+## CRD Validation Ratcheting
+    
+This Allows CRs to fail validation if the patch did not alter any of the invalid fields.  The ability to shift left validation logic from controllers to the front end is a long-term goal for improving the useability of the Kubernetes project.
+    
+## Add Generic Control Plane Staging Repositories
+
+This factors the kube-apiserver to build on a new staging repository that consumes k/apiserver but has a bigger, carefully chosen subset of the functionality of kube-apiserver such that it is reusable.
+
+The factoring will be progressive: we will start with a new repo that adds
+nothing to k/apiserver, and then progressively move generic functionality from
+kube-apiserver to the new repository. The new repo will be named
+k/generic-controlplane.
+  
+## CDI Support To The Device Plugin API
+
+CDI provides a standardized way of injecting complex devices into a container (i.e. devices that logically require more than just a single /dev node to be injected for them to work). This new feature enables plugin developers to utilize the CDIDevices field added to the CRI in 1.27 to pass CDI devices directly to CDI enabled runtimes (of which containerd and crio-o are in recent releases).
+    
+## Sidecar Containers Built-in Support
+
+This introduces a restartPolicy field to init containers and uses it to indicate that an init container is a sidecar container. Kubelet will start init containers with restartPolicy=Always in the order with other init containers, but instead of waiting for its completion, it will wait for the container startup completion.
```

Comment:
```suggestion
Kubernetes 1.28 introduces an alpha `restartPolicy` field for [init containers](/docs/concepts/workloads/pods/init-containers/),
and uses that to indicate when an init container is also a _sidecar container_. The will start init containers with `restartPolicy: Always` in the order they are defined, along with other init containers. Instead of waiting for that sidecar container to complete before starting the main container(s) for the Pod, the kubelet only waits for
the sidecar init container to have started.
```